Automattic/wp-calypso
# Plugins: add error notice explanation for a server error

Related to https://github.com/Automattic/wp-calypso/issues/92251

## Proposed Changes

* Adds an error notice explanation when the plugin install experiences a server error when activating the plugin
* Adds a partial success action to allow us to handle case where install was successful but the activation failed. This will allow the modal to reflect this by showing the installed (but deactivated) plugin.

https://github.com/Automattic/wp-calypso/assets/5560595/ee2ce058-9f9f-4982-bc6c-1bf971fe9e0b



## Before

<img width="629" alt="Screenshot 2024-07-02 at 14 26 51" src="https://github.com/Automattic/wp-calypso/assets/5560595/251b7fa6-4db8-44c3-96a0-1f7be20a99be">

## After

<img width="1255" alt="Screenshot 2024-07-02 at 14 28 19" src="https://github.com/Automattic/wp-calypso/assets/5560595/f0ea55a1-04bb-46c1-9eed-3b24cdf54b8c">
<img width="1471" alt="Screenshot 2024-07-03 at 18 29 42" src="https://github.com/Automattic/wp-calypso/assets/5560595/f27f4c01-f856-418a-bc39-5c7b595f4702">

## Why are these changes being made?


* Better explanation to user what happened when they tried to install a plugin

## Testing Instructions



1. Go to global plugins view at `/plugins/widont-part-deux`
2. Install the plugin on a WoA site
3. Observe the install and eventual error message
4. Go to `wp-admin` and verify that the plugin did install but did not activate.


## Pre-merge Checklist
